test/avrora/codec/plain_test.exs

```diff
@@ -154,17 +165,26 @@ defmodule Avrora.Codec.PlainTest do
 
   defp payment_payload, do: %{"id" => "00000000-0000-0000-0000-000000000000", "amount" => 15.99}
 
+  defp null_payment_payload,
+    do: %{"id" => "00000000-0000-0000-0000-000000000000", "amount" => nil}
+
   defp payment_schema do
     {:ok, schema} = Schema.parse(payment_json_schema())
     %{schema | id: nil, version: nil}
   end
 
   defp payment_json_schema do
-    ~s({"namespace":"io.confluent","name":"Payment","type":"record","fields":[{"name":"id","type":"string"},{"name":"amount","type":"double"}]})
+    ~s({"namespace":"io.confluent","name":"Payment","type":"record","fields":[{"name":"id","type":"string"},{"name":"amount","type":["null","double"]}]})
```

Comment:
Since you already introduce null_payment_payload, null_payment_message, I would suggest going with another schema (most of the things already different). It will also allow us to avoid confusion from the logical side (nullable amount).